features/netsocket/UDPSocket.cpp

```diff
@@ -126,6 +140,12 @@ nsapi_size_or_error_t UDPSocket::recvfrom(SocketAddress *address, void *buffer,
     return ret;
 }
 
+nsapi_size_or_error_t UDPSocket::recv(void *buffer, nsapi_size_t size)
+{
+    SocketAddress ignored; // Dangerous, I'm spending ~50 bytes from stack for address space that I'll ignore.
+    return recvfrom(&ignored, buffer, size);
```

Comment:
Do we believe NSAPI stacks won't cope with being passed `NULL`? If so, this dummy actually needs to be in `UDPSocket::recvfrom`, as we want `recv` to be equivalent to `recvfrom(NULL)`.